linera-views/src/hashable_wrapper.rs

```diff
@@ -8,13 +8,17 @@ use crate::{
 };
 use async_lock::Mutex;
 use async_trait::async_trait;
+use futures::join;
 use serde::{de::DeserializeOwned, Serialize};
-use std::ops::{Deref, DerefMut};
+use std::{
+    marker::PhantomData,
+    ops::{Deref, DerefMut},
+};
 
 /// A hash for ContainerView and storing of the hash for memoization purposes
 #[derive(Debug)]
 pub struct WrappedHashableContainerView<C, W, O> {
-    context: C,
+    _phantom: PhantomData<C>,
     stored_hash: Option<O>,
     hash: Mutex<Option<O>>,
     inner: W,
```

Comment:
I am not sure if I understand the question right.
We certainly do not want the inner to memorize the hashes as that would be a pure duplication of data.
And no we do not want the inner to be a WrappedContainerView as that would be a recursive construction. Without the possibility of getting out of it with an `Option` or similar, this would get us immediate memory error if that even compiles.

Reply:
(Ah, you're right. I actually meant that if `inner` is a collection of views, can we enforce statically that _those_ views memoize their hashes. But we probably can't.)